src/plugins/console/public/lib/autocomplete/autocomplete.ts

```diff
@@ -349,6 +349,47 @@ export default function ({
     });
   }
 
+  function getConditionalTemplate(term: { value?: string; context?: AutoCompleteContext }) {
+    const context = term.context;
+    if (context?.endpoint && term.value) {
+      const { data_autocomplete_rules: autocompleteRules } = context.endpoint;
+      if (autocompleteRules && autocompleteRules[term.value]) {
+        const currentLineNumber = editor.getCurrentPosition().lineNumber;
+        const rules = (autocompleteRules[term.value] as any).__one_of ?? [];
+        const startLine = getStartLineNumber(currentLineNumber, context.requestStartRow!, rules);
```

Comment:
You are taking the request start row but should take the start row of the block. It can be different if it is a nested block (object in the array, for example). Please, implement the required logic.

Reply:
`requestStartRow` is a fallback param to break out of recursion. It stops at line that that matches the condition within the given block. This will handle nested blocks as well

Reply:
Agree, we don't need the reqeustStartRow here, this takes the line that matches the condition in the block. It works for nested blocks as well